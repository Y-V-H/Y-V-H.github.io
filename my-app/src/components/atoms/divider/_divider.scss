@import '../../_settings/variables';

.a-divider {
    border: 1px solid $raisin-black;
    height: 1px;
    background: $dark-liver;
}