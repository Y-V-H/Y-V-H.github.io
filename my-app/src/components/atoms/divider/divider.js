import React from 'react';
import PropTypes from 'prop-types';

export const Divider = ({ className }) => <hr className={`a-divider ${className ? className : ''}`} />


Divider.propTypes = {
    className: PropTypes.string,
}