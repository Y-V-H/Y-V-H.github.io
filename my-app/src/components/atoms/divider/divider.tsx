import React from 'react';

type DividerProps = {
    className?: string
}

export const Divider = ({ className }: DividerProps) => <hr className={`a-divider ${className ? className : ''}`} />
