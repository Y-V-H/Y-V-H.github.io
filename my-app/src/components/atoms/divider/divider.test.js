import renderer from 'react-test-renderer';
import React from 'react';
import { Divider } from './divider';


test('should be render', () => {
    const component = renderer.create(
        <Divider />
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('should be render with custom class', () => {
    const component = renderer.create(
        <Divider className='test'/>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});