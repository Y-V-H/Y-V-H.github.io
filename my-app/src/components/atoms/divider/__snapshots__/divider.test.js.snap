// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be render 1`] = `
<hr
  className="a-divider undefined"
/>
`;

exports[`should be render with custom class 1`] = `
<hr
  className="a-divider test"
/>
`;
